android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -1543,16 +1546,13 @@ public static void getTxExtraInfo(BraveWalletBaseActivity activity, NetworkInfo
                 });
     }
 
-    public static void getP3ANetworks(Callbacks.Callback1<NetworkInfo[]> callback) {
-        BraveActivity activity = BraveActivity.getBraveActivity();
-        if (activity == null) {
-            callback.call(new NetworkInfo[0]);
-        }
+    public static void getP3ANetworks(WalletModel walletModel,
+            Callbacks.Callback1<NetworkInfo[]> callback) {
         ArrayList<NetworkInfo> relevantNetworks = new ArrayList<NetworkInfo>();
         boolean countTestNetworks = CommandLine.getInstance().hasSwitch(
                 BraveWalletConstants.P3A_COUNT_TEST_NETWORKS_SWITCH);
         LiveDataUtil.observeOnce(
-                activity.getWalletModel().getCryptoModel().getNetworkModel().mCryptoNetworks,
+                walletModel.getCryptoModel().getNetworkModel().mCryptoNetworks,
```

Comment:
perhaps a good idea to make a check on `walletModel == null` as it could be null when an activity where we get it is null. And call `callback.call(new NetworkInfo[0]);` and return after that.

Reply:
Updated with Pavneet's suggestion, so now it is passed with `NetworkInfo[]` which cannot be `null`.

Reply:
Still I see null checks missed in other places.